### PR TITLE
[BugFix]Avoid BE crash when ORC's type can't convert to SlotDescriptor's type

### LINKED ISSUE #14156: [HIVE-METADATA] BE crash from hive external table after recreated hive table
### Steps to reproduce the behavior (Required)
```
create external catalog hive_catalog_77227724_6d96_11ed_baf6_00163e0e550b PROPERTIES(
                "type" = "hive",
                "hive.metastore.uris" = "thrift://172.26.194.238:9083",
                "enable_hive_metastore_cache" = "false",
                "enable_remote_file_cache" = "false");
create external resource hive_res_77261550_6d96_11ed_baf6_00163e0e550b PROPERTIES(
                "type" = "hive",
                "hive.metastore.uris" = "thrift://172.26.194.238:9083",
                "enable_hive_metastore_cache" = "false",
                "enable_remote_file_cache" = "false");
create table hive_hdfs_orc(
                     col_tinyint tinyint comment "column tinyint"
                    ,col_decimal decimal(10,2) comment "column decimal"
                    ,col_timestamp timestamp comment "column timestamp"
                    ,col_date date comment "column date"
                    ,col_string string comment "column string"
                    ,col_boolean boolean comment "column boolean"
                    ) stored as orc
create external table extbl_hive_hdfs_orc(
                     col_tinyint tinyint comment "column tinyint"
                    ,col_decimal decimal(10,2) comment "column decimal"
                    ,col_timestamp datetime comment "column timestamp"
                    ,col_date date comment "column date"
                    ,col_string string comment "column string"
                    ,col_boolean boolean comment "column boolean")
                    ENGINE=hive
                    properties ("resource" = "hive_res_77261550_6d96_11ed_baf6_00163e0e550b", "table" = "hive_hdfs_orc", "database" = "hive_metadata_test_772275c6_6d96_11ed_baf6_00163e0e550b");
```

```
drop table hive_hdfs_orc
select * from hive_metadata_77207dac_6d96_11ed_baf6_00163e0e550b.extbl_hive_hdfs_orc order by 1;
execute sql error and messages: (1064, 'Failed to get partition keys on [hive_metadata_test_772275c6_6d96_11ed_baf6_00163e0e550b.hive_hdfs_orc] from meta store: Internal error processing get_partition_names'), and time is: 2022-11-26 22:28:55
select * from hive_catalog_77227724_6d96_11ed_baf6_00163e0e550b.hive_metadata_test_772275c6_6d96_11ed_baf6_00163e0e550b.hive_hdfs_orc;
execute sql error and messages: (1064, 'Failed to get partition keys on [hive_metadata_test_772275c6_6d96_11ed_baf6_00163e0e550b.hive_hdfs_orc] from meta store: Internal error processing get_partition_names'), and time is: 2022-11-26 22:28:56
create table hive_hdfs_orc(
                     col_tinyint tinyint comment "column tinyint"
                    ,col_date date comment "column date"
                    ,col_string string comment "column string"
                    ,col_array array<int>
                    ) stored as orc
insert into hive_hdfs_orc select 100,'2025-10-15','Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?',array(1,11,111)
insert into hive_hdfs_orc select 101,'2007-06-07','Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.',array(2,22,222)
select * from hive_hdfs_orc order by col_tinyint
2 rows in hive table
(100, '2025-10-15', 'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?', '[1,11,111]')
(101, '2007-06-07', 'Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.', '[2,22,222]')
select * from hive_metadata_77207dac_6d96_11ed_baf6_00163e0e550b.extbl_hive_hdfs_orc order by 1;
execute sql error and messages: (1064, 'A error occurred: errorCode=2001 errorMessage:Channel inactive error!, host: unknown'), and time is: 2022-11-26 22:29:52
refresh external table hive_metadata_77207dac_6d96_11ed_baf6_00163e0e550b.extbl_hive_hdfs_orc;
select * from hive_metadata_77207dac_6d96_11ed_baf6_00163e0e550b.extbl_hive_hdfs_orc order by 1;
execute sql error and messages: (1064, 'rpc failed, host: 172.26.92.201'), and time is: 2022-11-26 22:29:52
```

be.out:
```
query_id:abef862a-2c1f-4f41-882b-54891cba9b9f, fragment_instance:abef862a-2c1f-4f41-882b-54891cba9ba1
*** Aborted at 1669472986 (unix time) try "date -d @1669472986" if you are using GNU date ***
PC: @          0x3ce11ab starrocks::vectorized::OrcMapping::get_column_id_or_child_mapping()
*** SIGSEGV (@0x8) received by PID 29493 (TID 0x7f9016fc6700) from PID 8; stack trace: ***
    @          0x5789a22 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f90b992e655 os::Linux::chained_handler()
    @     0x7f90b9933481 JVM_handle_linux_signal
    @     0x7f90b9926678 signalHandler()
    @     0x7f90b8e5b630 (unknown)
    @          0x3ce11ab starrocks::vectorized::OrcMapping::get_column_id_or_child_mapping()
    @          0x3cd20bf starrocks::vectorized::_create_type_descriptor_by_orc()
    @          0x3cd2f4a starrocks::vectorized::OrcChunkReader::_init_src_types()
    @          0x3cd9fc3 starrocks::vectorized::OrcChunkReader::init()
    @          0x4bcfade starrocks::vectorized::HdfsOrcScanner::do_open()
    @          0x4bc92ba starrocks::vectorized::HdfsScanner::open()
    @          0x4b4811c starrocks::connector::HiveDataSource::_init_scanner()
    @          0x4b4a205 starrocks::connector::HiveDataSource::open()
    @          0x2e7b9d9 starrocks::pipeline::ConnectorChunkSource::_read_chunk()
    @          0x2e6575c starrocks::pipeline::ChunkSource::buffer_next_batch_chunks_blocking()
    @          0x2b416e4 _ZZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS_12RuntimeStateEiENKUlvE_clEv
    @          0x2b5259d starrocks::workgroup::ScanExecutor::worker_thread()
    @          0x4635245 starrocks::ThreadPool::dispatch_thread()
    @          0x463065a starrocks::Thread::supervise_thread()
    @     0x7f90b8e53ea5 start_thread
    @     0x7f90ae539b0d __clone
    @                0x0 (unknown)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.5-RELEASE 7f0e43b |
+----------------------------+
1 row in set (0.01 sec)
```

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14156

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 1]
```diff
@@ -49,9 +49,9 @@ const static std::unordered_map<orc::TypeKind, LogicalType> g_orc_starrocks_type
 };
 
 // NOLINTNEXTLINE
-const static std::set<orc::TypeKind> g_orc_int_type = {
```
Comment (stephen-shelby) [inline be/src/formats/orc/orc_chunk_reader.cpp:52]: delete it

Comment (Smith-Cruise) [reply be/src/formats/orc/orc_chunk_reader.cpp:52]: fixed

**Review by dirtysalt — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

File: be/src/formats/orc/orc_mapping.cpp [hunk 2]
```diff
@@ -117,6 +117,26 @@ StatusOr<std::unique_ptr<OrcMapping>> OrcMappingFactory::build_mapping(
     return orc_mapping;
 }
 
+Status OrcMappingFactory::_check_orc_type_can_converte_2_logical_type(const orc::Type& orc_source_type,
+                                                                      const TypeDescriptor& slot_target_type) {
+    bool can_convert = true;
```
Comment (zuyu) [inline be/src/formats/orc/orc_mapping.cpp:122]: This should be `false`, right?

Comment (Smith-Cruise) [reply be/src/formats/orc/orc_mapping.cpp:122]: We only check for complex types now, if `orc_source_type` is a primitive type, we will not check it. So we need set `can_convert=true` by default.

**Pull request merged**